### PR TITLE
fix: Show JS Function data in autocompletion hints

### LINKED ISSUE #15909: [Bug]-[40000]:.data does not appear as an autocomplete suggestion when using JS functions in a binding
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

.data does not appear as an autocomplete suggestion when using JS functions in a binding

### Steps To Reproduce

1. Use a JS function in a binding like `JSObject1.myFunc1.` and observe that data does not appear as a suggestion on autocomplete

image

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

This PR adds JS function data to autocompletion hints

Fixes #15909 

Screenshot 2023-01-16 at 20 35 55

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/utils/autocomplete/dataTreeTypeDefCreator.ts [hunk 1]
```diff
@@ -67,15 +68,15 @@ export const dataTreeTypeDefCreator = (
...
-      for (const key in metaObj) {
-        // const jsFunctionObj = metaObj[key];
-        // const { arguments: args } = jsFunctionObj;
-        // const argsTypeString = getFunctionsArgsType(args);
-        // As we don't show args we avoid to get args def of function
-        // we will also need to check performance implications here
-
-        const argsTypeString = getFunctionsArgsType([]);
-        jsPropertiesDef[key] = argsTypeString;
+      for (const funcName in metaObj) {
```
Comment (arunvjn) [inline app/client/src/utils/autocomplete/dataTreeTypeDefCreator.ts:71]: Does this take of the possibility of nested functions ?
```
{
 myVar1: { 
   fn1: () => Api1.data
 }
 myFn2: () => Api1.data
}
```

Comment (arunvjn) [reply app/client/src/utils/autocomplete/dataTreeTypeDefCreator.ts:71]: Nevermind, this case was never working anyway

**Comment by ramsaptami** [comment]
Tested autocomplete suggestions for `.data` accessor for a JS function. We're gtg to release @ohansFavour

**Pull request merged**